File: 937 - Reorder Data in Log Files.py
class Solution:
    def reorderLogFiles(self, logs: list[str]) -> list[str]:
        digit, word = [], []
        for log in logs:
            arr = log.split()
            if arr[1].isdigit():
                digit.append(log)
            else:
                word.append(arr)
        word = sorted(word, key= lambda a : (a[1:], a[0]))
        res = []
        for log in word:
            res.append(' '.join(log))
        res = res + digit
        return res

File: 937 - Reorder Data in Log Files.js
/**
 * @param {string[]} logs
 * @return {string[]}
 */
var reorderLogFiles = function (logs) {
    let digit = [], letter = [];
    let patt = /\d+/;
    for (let log of logs) {
        let arr = log.split(' ');
        if (patt.test(arr[1])) {
            digit.push(log)
        } else {
            letter.push(log)
        }
    }
    letter.sort();
    letter.sort((a, b) => a.slice(a.indexOf(' ')) < b.slice(b.indexOf(' ')) ? -1 : 1);
    return letter.concat(digit);
};